### PR TITLE
add GRASS r.path

### LINKED ISSUE #44125: r.path not contained in Processing toolbox
**Describe the bug**
The Grass tool [r.path](https://grass.osgeo.org/grass78/manuals/r.path.html) is missed in the GRASS toolbox.

**How to Reproduce**
1. Menu: Processing/Toolbox
2. Enter r.path in the search field

Tool r.path is not found even though its a valid tool ofGRASS 7.8

**QGIS and OS versions**
3.16.8-4 on Windows 10

**Additional remarks**
- The executable is contained in C:\Program Files\QGIS 3.16.8\apps\grass\grass78\bin: r.path.exe
- It seems the tool was forgotten to add to the toolbox
- This is a qGis problem and not a GRASS plugin problem

### PR DESCRIPTION
**Pull request by gioman**
fixes https://github.com/qgis/QGIS/issues/44125

### CODE REVIEW

File: python/plugins/grassprovider/description/r.path.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+r.path
+Traces paths from starting points following input directions.
+Raster (r.*)
+QgsProcessingParameterRasterLayer|input|Name of input direction
+QgsProcessingParameterEnum|format|Format of the input direction map|auto;degree;45degree;bitmask|false|0|false
+QgsProcessingParameterRasterLayer|values|Name of input raster values to be used for output|None|True
+QgsProcessingParameterRasterDestination|raster_path|Name for output raster path map
+QgsProcessingParameterVectorDestination|vector_path|Name for output vector path map
+QgsProcessingParameterPoint|start_coordinates|Map coordinates of starting point(s) (E,N)|None|True
```
Comment (alexbruy) [inline python/plugins/grassprovider/description/r.path.txt:9]: Looking at the GRASS docs it seems that this parameter can accept multiple points, so it uses plural for "coordinates". Maybe it is better to change parameter description to reflect that in Processing tool only single point is accepted?

File: python/plugins/grassprovider/description/r.path.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+QgsProcessingParameterFeatureSource|start_points|Vector layer containing starting point(s)|0|None|True
```
Comment (alexbruy) [inline python/plugins/grassprovider/description/r.path.txt:10]: Better to set allowed geometry type to points.

Also probably it is worth to split the tool into two: one accepting point and another one accepting vector layer with starting points?

**Comment by alexbruy** [comment]
Can you please add tests too?

File: python/plugins/grassprovider/description/r.path.txt (hunk 2 above)
Comment (gioman) [reply python/plugins/grassprovider/description/r.path.txt:10]: @alexbruy 



> Better to set allowed geometry type to points.

isn't 0 already correct for points? It does not list lines or polygons here with 0.

Comment (gioman) [reply python/plugins/grassprovider/description/r.path.txt:9]: > Also probably it is worth to split the tool into two: one accepting point and another one accepting vector layer with starting points?

done

File: python/plugins/grassprovider/description/r.path.txt (hunk 1 above)
Comment (gioman) [reply python/plugins/grassprovider/description/r.path.txt]: > Looking at the GRASS docs it seems that this parameter can accept multiple points, so it uses plural for "coordinates". Maybe it is better to change parameter description to reflect that in Processing tool only single point is accepted?

done

**Comment by gioman** [comment]
> Can you please add tests too?

will try

**Comment by gioman** [comment]
Not stale.

**Comment by nyalldawson** [comment]
@gioman 
> Can you please add tests too?

Any updates on that? Would be good to get this merged so that we can cleanup the queue...

**Comment by gioman** [comment]
> Any updates on that? Would be good to get this merged so that we can cleanup the queue...

yes, will try do it asap.

**Comment by alexbruy** [comment]
Superseded by #44974.

**Comment by gioman** [comment]
@alexbruy thanks!

**Pull request closed without merging**